### ISSUE TITLE
fish.py 0.9.3 error

### ISSUE THREAD

**Issue by dr0id123** [weechat/scripts#384]
## Script / WeeChat

- Name of script: fish.py
- Script version: 0.9.3
- WeeChat version:  2.6

This bug has been reported to the script author? No
Response or reason: 

## Bug summary

** Cannnot view encrypted channel **

python: stdout/stderr (fish): Traceback (most recent call last):
01:41:36 PM		python: stdout/stderr (fish):   File "/home/XXXX/.weechat/python/autoload/fish.py", line 313, in blowcrypt_unpack
01:41:36 PM		python: stdout/stderr (fish):     plain = cipher.decrypt(raw)
01:41:36 PM		python: stdout/stderr (fish):   File "/home/XXXX/.weechat/python/autoload/fish.py", line 238, in decrypt
01:41:36 PM		python: stdout/stderr (fish):     return self.blowfish.decrypt(data)
01:41:36 PM		python: stdout/stderr (fish):   File "/usr/lib/python3/dist-packages/Crypto/Cipher/blockalgo.py", line 295, in decrypt
01:41:36 PM		python: stdout/stderr (fish):     return self._cipher.decrypt(ciphertext)
01:41:36 PM		python: stdout/stderr (fish): ValueError: Input strings must be a multiple of 8 in length
01:41:36 PM		python: stdout/stderr (fish): During handling of the above exception, another exception occurred:
01:41:36 PM		python: stdout/stderr (fish): Traceback (most recent call last):
01:41:36 PM		python: stdout/stderr (fish):   File "/home/XXXX/.weechat/python/autoload/fish.py", line 707, in fish_modifier_in_privmsg_cb
01:41:36 PM		python: stdout/stderr (fish):     clean = blowcrypt_unpack(match.group(5), b)
01:41:36 PM		python: stdout/stderr (fish):   File "/home/XXXX/.weechat/python/autoload/fish.py", line 315, in blowcrypt_unpack
01:41:36 PM		python: stdout/stderr (fish):     raise MalformedError
01:41:36 PM		python: stdout/stderr (fish): NameError: name 'MalformedError' is not defined
01:41:36 PM	=!=	python: error in function "fish_modifier_in_privmsg_cb"

VERSION of Python:
python3 --version
Python 3.6.9

OS:
Ubuntu 18.04.3 LTS

**Comment by flashcode** [comment]
Are you using Python 3 in WeeChat? Please show output of `/python version`.
(You can have both Python 2 and 3 on your system, but only one of them is used in WeeChat).

**Comment by dr0id123** [comment]
output of /python version:

python: 3.6.5

**Comment by flashcode** [comment]
For now, this script has not yet been ported to Python 3, so it can work only with Python 2.
See the Python 3 transition page for more info: https://weechat.org/scripts/python3/
Authors are warned since June 2018 that scripts should be compatible with Python 3, and since July 2019, WeeChat is compiled with Python 3 by default.

So I suggest to ask the author of the script to make it compatible with Python 2 + 3, which is now urgent: at the end of year, all scripts not compatible with Python 3 will be disabled.

**Comment by dr0id123** [comment]
Leaving this for all that stumble along the same issue -->

It's hard to know if/when fish.py will be updated to python3 as it appears the maintainer isn't active.  As an alternative, there is a ruby implementation of this functionality available in the weechat repository.

**Steps I took:**

1. Remove fish.py plugin from weechat.
2. Download/install this ruby script: https://weechat.org/scripts/source/weefish.rb.html/
3. **On Ubuntu 18**, there is an error with the above ruby script loading the library "crypt/blowfish".  To fix this issue, I had to install the appropriate ruby gem, then ensure that gem was available in a location of which ruby looks for libraries (in other words, it needs to be in the right path....) *sigh*.  
```
sudo gem install crypt
sudo cp -R /var/lib/gems/2.5.0/gems/crypt-2.2.1/lib/crypt /usr/lib/ruby/vendor_ruby/
```
4. Reload weechat.  Go to to the channel buffer and do `/setkey <yourkeyhere>`, then `/save` to write the configuration disk.

Hope this helps!

**Comment by nikster2** [comment]
I've managed to make a working version if the script for python 3+. Created a pull request which anyone interested and try. Works for me, but won't give any support if it doesn't work for you ;) Thought I would just share my try at fixing it as a base for a final solution.

#398

**Comment by flashcode** [comment]
The version 0.9.4 of the script has been released (via PR #406), so the script should be compatible with Python 3 now.
I close this issue.
